app/services/notifications/remove_each.rb

```diff
@@ -0,0 +1,27 @@
+module Notifications
+  class RemoveEach
+    def initialize(notifiable_collection, action)
+      @notifiable_collection = notifiable_collection
+      @action = action
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      # only used for mentions since it's an array
+      notifiable_collection.each do |notifiable|
```

Comment:
here you can call `notifiable_collection.select(:id).find_each`. The reason being that you end up loading less data from the DB and it's "future proof" because with `find_each` at worst you batch the load. It's just a suggestion, AR is full of little tricks like this :D

Reply:
Thank you, I had not though about that this way. Though I did some refactoring, since the call was not only for Mention only anymore. And I pass hashes instead on objects, so I guess all unneeded stuff is not there anymore.